signer/src/storage/postgres.rs

```diff
@@ -687,6 +687,94 @@ impl super::DbWrite for PgStore {
         Ok(())
     }
 
+    async fn write_deposit_requests(
+        &self,
+        deposit_requests: Vec<model::DepositRequest>,
+    ) -> Result<(), Self::Error> {
+        if deposit_requests.is_empty() {
+            return Ok(());
+        }
+
+        let mut txid = Vec::with_capacity(deposit_requests.len());
+        let mut output_index = Vec::with_capacity(deposit_requests.len());
+        let mut spend_script = Vec::with_capacity(deposit_requests.len());
+        let mut reclaim_script = Vec::with_capacity(deposit_requests.len());
+        let mut recipient = Vec::with_capacity(deposit_requests.len());
+        let mut amount = Vec::with_capacity(deposit_requests.len());
+        let mut max_fee = Vec::with_capacity(deposit_requests.len());
+        let mut sender_addresses = Vec::with_capacity(deposit_requests.len());
+
+        for req in deposit_requests {
+            txid.push(req.txid);
+            output_index.push(req.output_index);
+            spend_script.push(req.spend_script);
+            reclaim_script.push(req.reclaim_script);
+            recipient.push(req.recipient);
+            amount.push(req.amount);
+            max_fee.push(req.max_fee);
+            // We need to join the addresses like this (and later split
+            // them), because handling of multi-dimensional arrays in
+            // postgres is tough. The naive approach of doing
+            // UNNEST($1::VARCHAR[][]) doesn't work, since that completely
+            // flattens the array.
+            sender_addresses.push(req.sender_addresses.join(","));
+        }
+
+        sqlx::query(
```

Comment:
Not for right now, but do we have any documentation on the structure of the SQL database? I find these very hard to follow and am really just trusting the unit tests here.

Reply:
Yeah we do. The migration query that created this table is here: https://github.com/stacks-network/sbtc/blob/dcbdb12183797cfe69067a789120992ab0cb4621/signer/migrations/0003__create_tables.sql#L27-L38

The rest are in that file.